chain/db.go

```diff
@@ -623,7 +625,11 @@ func NewDBStore(db DB, n *consensus.Network, genesisBlock types.Block) (_ *DBSto
 		return nil, Checkpoint{}, errors.New("detected siad database, refusing to proceed")
 	}
 
-	dbs := &DBStore{db: db, n: n}
+	dbs := &DBStore{
+		db:  db,
+		n:   n,
+		enc: &types.Encoder{},
```

Comment:
I think we can use a non-pointer for the `enc` field; that way, we don't need to explicitly initialize it here